crates/types/src/rpc/contract_class.rs

```diff
@@ -198,25 +193,114 @@ fn convert_sierra_to_codegen(
     })
 }
 
-impl TryInto<CodegenContractClass> for ContractClass {
-    type Error = Error;
-    fn try_into(self) -> Result<CodegenContractClass, Self::Error> {
-        match self {
-            ContractClass::Cairo0(contract_class) => {
-                Ok(CodegenContractClass::Legacy(contract_class.try_into()?))
-            }
-            ContractClass::Cairo1(contract_class) => {
-                Ok(CodegenContractClass::Sierra(convert_sierra_to_codegen(&contract_class)?))
-            }
-        }
+pub fn compute_sierra_class_hash(contract_class: &SierraContractClass) -> DevnetResult<Felt> {
+    let mut contract_class_json_value =
+        serde_json::to_value(contract_class).map_err(JsonError::SerdeJsonError)?;
+
+    // to match SierraClass struct, the field sierra_program_debug_info dont have to be
+    // Option::None, because during serialization it gets converted to null
+    // and the next deserialzation to SierraClas will fail, because it expects this key to have some
+    // value
+    if contract_class.sierra_program_debug_info.is_none() {
+        contract_class_json_value["sierra_program_debug_info"] =
+            serde_json::to_value(SierraClassDebugInfo {
+                type_names: Default::default(),
+                libfunc_names: Default::default(),
+                user_func_names: Default::default(),
+            })
+            .map_err(JsonError::SerdeJsonError)?;
+    }
+
+    let sierra_class: SierraClass =
+        serde_json::from_value(contract_class_json_value).map_err(JsonError::SerdeJsonError)?;
+
+    Ok(sierra_class
+        .class_hash()
+        .map_err(|_| Error::ConversionError(ConversionError::InvalidFormat))?
+        .into())
+}
+
+/// Computes cairo_lang_starknet::CasmContractClass hash.
+/// Implementation copied from starknet_in_rust
+/// # Arguments
+///
+/// * `contract_class` - The contract class in casm format.
+pub fn compute_casm_class_hash(contract_class: &CasmContractClass) -> DevnetResult<Felt> {
```

Comment:
Why did we copy this function (it's only used in tests) if we are keeping starknet_in_rust as a dependency?